src/drive/mobile/modules/upload/index.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react'
+import localforage from 'localforage'
+import { compose } from 'redux'
+import { connect } from 'react-redux'
+import { withRouter } from 'react-router'
+import PropTypes from 'prop-types'
+
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import { Query, cancelable, withClient } from 'cozy-client'
+
+import { translate } from 'cozy-ui/transpiled/react'
+import { Modal } from 'cozy-ui/transpiled/react'
+
+import { uploadFilesFromNative } from 'drive/web/modules/upload'
+
+import { ROOT_DIR_ID, TRASH_DIR_ID } from 'drive/constants/config'
+import Header from 'drive/web/modules/move/Header'
+import Explorer from 'drive/web/modules/move/Explorer'
+import FileList from 'drive/web/modules/move/FileList'
+import Loader from 'drive/web/modules/move/Loader'
+import LoadMore from 'drive/web/modules/move/LoadMore'
+import Footer from 'drive/web/modules/move/Footer'
+import Topbar from 'drive/web/modules/move/Topbar'
+
+export const generateForQueue = files => {
+  const filesForQueue = []
+  files.map(file => {
+    filesForQueue.push({ file: file, isDirectory: false })
+  })
+  return filesForQueue
+}
+
+export class DumbUpload extends Component {
+  state = {
+    items: null,
+    folderId: ROOT_DIR_ID
+  }
+  async componentDidMount() {
+    const items = await localforage.getItem('importedFiles')
+    this.setState({ items })
+  }
+
+  async uploadAndState() {
+    const { items, folderId } = this.state
+    const { router, uploadFilesFromNative } = this.props
+    const filesForQueue = generateForQueue(items)
+    uploadFilesFromNative(filesForQueue, folderId, this.callbackSuccess)
+    setTimeout(() => router.push(`/folder/${folderId}`), 500)
```

Comment:
Why the arbitrary 500ms timeout?

Reply:
We can discuss about the 500, and I think 50 is ok. Will change it.

The setTimeout() is here to be sure that the first dispatch was done. I think that during my tests I had a concurrent action resulting in errors... Will add a comment

Reply:
(Comment or `const NO_CONCURRENT_ACTION_DELAY = 500`, or both ?)